src/pages/rekey-reason/rekey-reason.reducer.ts

```diff
@@ -1,47 +1,47 @@
 import * as testActions from './../../modules/tests/tests.actions';
-import * as rekeyReasonActions from './rekey-reason.actions';
+import * as rekeyActions from './../../modules/tests/rekey/rekey.actions';
 import { createFeatureSelector } from '@ngrx/store';
 import { RekeyReasonModel } from './rekey-reason.model';
+import { HttpStatusCodes } from '../../shared/models/http-status-codes';
 
 export const initialState: RekeyReasonModel = {
   uploadStatus: {
     isUploading: false,
     hasUploadSucceeded: false,
     hasUploadFailed: false,
+    isDuplicate: false,
   },
 };
 
-export function rekeyReasonReducer(state = initialState, action: testActions.Types | rekeyReasonActions.Types) {
+export function rekeyReasonReducer(state = initialState, action: testActions.Types | rekeyActions.Types) {
   switch (action.type) {
-    case rekeyReasonActions.REKEY_REASON_VIEW_DID_LEAVE:
+    case rekeyActions.END_REKEY:
       return {
         ...initialState,
       };
     case testActions.SEND_CURRENT_TEST:
       return {
         ...state,
         uploadStatus: {
+          ...initialState.uploadStatus,
           isUploading: true,
-          hasUploadSucceeded: false,
-          hasUploadFailed: false,
         },
       };
     case testActions.SEND_CURRENT_TEST_SUCCESS:
       return {
         ...state,
         uploadStatus: {
-          isUploading: false,
+          ...initialState.uploadStatus,
           hasUploadSucceeded: true,
-          hasUploadFailed: false,
         },
       };
     case testActions.SEND_CURRENT_TEST_FAILURE:
       return {
         ...state,
         uploadStatus: {
-          isUploading: false,
-          hasUploadSucceeded: false,
+          ...initialState.uploadStatus,
           hasUploadFailed: true,
+          isDuplicate: action.error && action.error.status === HttpStatusCodes.CONFLICT,
```

Comment:
Would be better to send the evaluation of this expression as an action payload

Reply:
Updated